### ISSUE TITLE
Strange behavior with local webpagetest/agent container

### ISSUE THREAD

**Issue by dekryptic** [sitespeedio/sitespeed.io#2970]
For some reason I have to run the sitespeed.io container twice to not get any errors when running local webpagetest containers. I've figured out it has something to do with the agent as I when I restart it from scratch the errors pop up again.

My docker-compose.json file:
```
{
  "version": "3.7",
  "services": {
    "wptServer": {
      "image": "webpagetest/server",
      "ports": ["4000:80"]
    },
    "wptAgent": {
      "image": "webpagetest/agent",
      "cap_add": ["SYS_ADMIN"],
      "init": true,
      "privileged": true,
      "environment": {
        "SERVER_URL": "http://localhost:4000/work/",
        "LOCATION": "Test",
        "EXTRA_ARGS": "--checknet yes"
      },
      "network_mode": "host"
    },
    "sitespeedio": {
      "image": "sitespeedio/sitespeed.io:12.6.0-plus1",
      "user": "root",
      "command": [
        "-n", "1",
        "--webpagetest.host", "http://localhost:4000",
        "--webpagetest.location", "Test",
        "--video",
        "https://sitespeed.io"
      ],
      "volumes": ["./logs:/sitespeed.io/sitespeed-result"],
      "environment": {},
      "network_mode": "host"
    }
  }
}
```

I run each command in a separate tab in my terminal, of course waiting for WPT to start until I run sitespeedio:
```
docker-compose -f docker-compose.json up wptServer
docker-compose -f docker-compose.json up wptAgent
docker-compose -f docker-compose.json up sitespeedio
```

Logs:
Run 1:
```
Attaching to wpt_sitespeedio_1
sitespeedio_1  | Google Chrome 81.0.4044.92
sitespeedio_1  | Mozilla Firefox 75.0
sitespeedio_1  | [2020-04-26 00:14:52] INFO: Versions OS: linux 4.4.0-1072-aws nodejs: v12.16.2 sitespeed.io: 12.6.0 browsertime: 8.6.1 coach: 5.0.1
sitespeedio_1  | [2020-04-26 00:14:54] INFO: Will run Lighthouse tests after Browsertime has finished
sitespeedio_1  | [2020-04-26 00:14:54] INFO: Sending url https://sitespeed.io to test on Page Speed Insights
sitespeedio_1  | [2020-04-26 00:14:54] INFO: Sending url https://sitespeed.io to test on http://localhost:4000
sitespeedio_1  | [2020-04-26 00:14:54] INFO: Running tests using Chrome - 1 iteration(s)
sitespeedio_1  | [2020-04-26 00:14:55] INFO: Testing url https://sitespeed.io iteration 1
sitespeedio_1  | [2020-04-26 00:15:01] INFO: Got https://sitespeed.io analysed from Google Page Speed Insights
sitespeedio_1  | [2020-04-26 00:15:04] ERROR: Failed waiting on page to finished loading, timed out after 300000 ms WebDriverError: chrome not reachable
sitespeedio_1  |   (Session info: chrome=81.0.4044.92)
sitespeedio_1  |   (Session info: chrome=81.0.4044.92)
sitespeedio_1  |     at Object.throwDecodedError (/usr/src/app/node_modules/selenium-webdriver/lib/error.js:550:15)
sitespeedio_1  |     at parseHttpResponse (/usr/src/app/node_modules/selenium-webdriver/lib/http.js:565:13)
sitespeedio_1  |     at Executor.execute (/usr/src/app/node_modules/selenium-webdriver/lib/http.js:491:26)
sitespeedio_1  |     at processTicksAndRejections (internal/process/task_queues.js:97:5)
sitespeedio_1  |     at async Driver.execute (/usr/src/app/node_modules/selenium-webdriver/lib/webdriver.js:700:17)
sitespeedio_1  | [2020-04-26 00:15:04] INFO: URL failed to load, trying 5 more time(s): Failed waiting on page to finished loading, timed out after 300000 ms
sitespeedio_1  | [2020-04-26 00:15:05] ERROR: Failed waiting on page to finished loading, timed out after 300000 ms WebDriverError: chrome not reachable
sitespeedio_1  |   (Session info: chrome=81.0.4044.92)
sitespeedio_1  |   (Session info: chrome=81.0.4044.92)
sitespeedio_1  |     at Object.throwDecodedError (/usr/src/app/node_modules/selenium-webdriver/lib/error.js:550:15)
sitespeedio_1  |     at parseHttpResponse (/usr/src/app/node_modules/selenium-webdriver/lib/http.js:565:13)
sitespeedio_1  |     at Executor.execute (/usr/src/app/node_modules/selenium-webdriver/lib/http.js:491:26)
sitespeedio_1  |     at processTicksAndRejections (internal/process/task_queues.js:97:5)
sitespeedio_1  |     at async Driver.execute (/usr/src/app/node_modules/selenium-webdriver/lib/webdriver.js:700:17)
sitespeedio_1  | [2020-04-26 00:15:05] INFO: URL failed to load, trying 4 more time(s): Failed waiting on page to finished loading, timed out after 300000 ms
sitespeedio_1  | [2020-04-26 00:15:06] ERROR: Failed waiting on page to finished loading, timed out after 300000 ms WebDriverError: chrome not reachable
sitespeedio_1  |   (Session info: chrome=81.0.4044.92)
sitespeedio_1  |   (Session info: chrome=81.0.4044.92)
sitespeedio_1  |     at Object.throwDecodedError (/usr/src/app/node_modules/selenium-webdriver/lib/error.js:550:15)
sitespeedio_1  |     at parseHttpResponse (/usr/src/app/node_modules/selenium-webdriver/lib/http.js:565:13)
sitespeedio_1  |     at Executor.execute (/usr/src/app/node_modules/selenium-webdriver/lib/http.js:491:26)
sitespeedio_1  |     at processTicksAndRejections (internal/process/task_queues.js:97:5)
sitespeedio_1  |     at async Driver.execute (/usr/src/app/node_modules/selenium-webdriver/lib/webdriver.js:700:17)
sitespeedio_1  | [2020-04-26 00:15:06] INFO: URL failed to load, trying 3 more time(s): Failed waiting on page to finished loading, timed out after 300000 ms
sitespeedio_1  | [2020-04-26 00:15:07] ERROR: Failed waiting on page to finished loading, timed out after 300000 ms WebDriverError: chrome not reachable
sitespeedio_1  |   (Session info: chrome=81.0.4044.92)
sitespeedio_1  |   (Session info: chrome=81.0.4044.92)
sitespeedio_1  |     at Object.throwDecodedError (/usr/src/app/node_modules/selenium-webdriver/lib/error.js:550:15)
sitespeedio_1  |     at parseHttpResponse (/usr/src/app/node_modules/selenium-webdriver/lib/http.js:565:13)
sitespeedio_1  |     at Executor.execute (/usr/src/app/node_modules/selenium-webdriver/lib/http.js:491:26)
sitespeedio_1  |     at processTicksAndRejections (internal/process/task_queues.js:97:5)
sitespeedio_1  |     at async Driver.execute (/usr/src/app/node_modules/selenium-webdriver/lib/webdriver.js:700:17)
sitespeedio_1  | [2020-04-26 00:15:07] INFO: URL failed to load, trying 2 more time(s): Failed waiting on page to finished loading, timed out after 300000 ms
sitespeedio_1  | [2020-04-26 00:15:08] ERROR: Failed waiting on page to finished loading, timed out after 300000 ms WebDriverError: chrome not reachable
sitespeedio_1  |   (Session info: chrome=81.0.4044.92)
sitespeedio_1  |   (Session info: chrome=81.0.4044.92)
sitespeedio_1  |     at Object.throwDecodedError (/usr/src/app/node_modules/selenium-webdriver/lib/error.js:550:15)
sitespeedio_1  |     at parseHttpResponse (/usr/src/app/node_modules/selenium-webdriver/lib/http.js:565:13)
sitespeedio_1  |     at Executor.execute (/usr/src/app/node_modules/selenium-webdriver/lib/http.js:491:26)
sitespeedio_1  |     at processTicksAndRejections (internal/process/task_queues.js:97:5)
sitespeedio_1  |     at async Driver.execute (/usr/src/app/node_modules/selenium-webdriver/lib/webdriver.js:700:17)
sitespeedio_1  | [2020-04-26 00:15:08] INFO: URL failed to load, trying 1 more time(s): Failed waiting on page to finished loading, timed out after 300000 ms
sitespeedio_1  | [2020-04-26 00:15:09] ERROR: Failed waiting on page to finished loading, timed out after 300000 ms WebDriverError: chrome not reachable
sitespeedio_1  |   (Session info: chrome=81.0.4044.92)
sitespeedio_1  |   (Session info: chrome=81.0.4044.92)
sitespeedio_1  |     at Object.throwDecodedError (/usr/src/app/node_modules/selenium-webdriver/lib/error.js:550:15)
sitespeedio_1  |     at parseHttpResponse (/usr/src/app/node_modules/selenium-webdriver/lib/http.js:565:13)
sitespeedio_1  |     at Executor.execute (/usr/src/app/node_modules/selenium-webdriver/lib/http.js:491:26)
sitespeedio_1  |     at processTicksAndRejections (internal/process/task_queues.js:97:5)
sitespeedio_1  |     at async Driver.execute (/usr/src/app/node_modules/selenium-webdriver/lib/webdriver.js:700:17)
sitespeedio_1  | [2020-04-26 00:15:09] INFO: URL failed to load, trying 0 more time(s): Failed waiting on page to finished loading, timed out after 300000 ms
sitespeedio_1  | [2020-04-26 00:15:09] ERROR: Could not load URL https://sitespeed.io UrlLoadError: Failed waiting on page to finished loading, timed out after 300000 ms
sitespeedio_1  |     at SeleniumRunner.wait (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:183:13)
sitespeedio_1  |     at processTicksAndRejections (internal/process/task_queues.js:97:5)
sitespeedio_1  | [2020-04-26 00:15:09] ERROR: UrlLoadError: Failed to load https://sitespeed.io
sitespeedio_1  |     at SeleniumRunner.loadAndWait (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:288:17)
sitespeedio_1  |     at processTicksAndRejections (internal/process/task_queues.js:97:5)
sitespeedio_1  | [2020-04-26 00:15:09] INFO: https://sitespeed.io
sitespeedio_1  | [2020-04-26 00:15:09] INFO: Start collecting Lighthouse result for https://sitespeed.io iteration 1
sitespeedio_1  | [2020-04-26 00:15:34] INFO: Got https://sitespeed.io analysed with id 200426_JJ_dafcb9732538909d37bb9e33b1084d6a from http://localhost:4000
sitespeedio_1  | [2020-04-26 00:15:34] INFO: WebPageTest result at: http://localhost:4000/results.php?test=200426_JJ_dafcb9732538909d37bb9e33b1084d6a
sitespeedio_1  | [2020-04-26 00:15:35] ERROR: Incomplete first view data for WPT test 200426_JJ_dafcb9732538909d37bb9e33b1084d6a, run 1
sitespeedio_1  | [2020-04-26 00:15:37] INFO: HTML stored in /sitespeed.io/sitespeed-result/sitespeed.io/2020-04-26-00-14-52
wpt_sitespeedio_1 exited with code 1
```
Run 2:
```
Starting wpt_sitespeedio_1 ... done
Attaching to wpt_sitespeedio_1
sitespeedio_1  | Google Chrome 81.0.4044.92
sitespeedio_1  | Mozilla Firefox 75.0
sitespeedio_1  | groupadd: group 'sitespeedio' already exists
sitespeedio_1  | useradd: user 'sitespeedio' already exists
sitespeedio_1  | [2020-04-26 00:16:57] INFO: Versions OS: linux 4.4.0-1072-aws nodejs: v12.16.2 sitespeed.io: 12.6.0 browsertime: 8.6.1 coach: 5.0.1
sitespeedio_1  | [2020-04-26 00:16:58] INFO: Will run Lighthouse tests after Browsertime has finished
sitespeedio_1  | [2020-04-26 00:16:58] INFO: Sending url https://sitespeed.io to test on Page Speed Insights
sitespeedio_1  | [2020-04-26 00:16:58] INFO: Sending url https://sitespeed.io to test on http://localhost:4000
sitespeedio_1  | [2020-04-26 00:16:59] INFO: Running tests using Chrome - 1 iteration(s)
sitespeedio_1  | [2020-04-26 00:16:59] INFO: Testing url https://sitespeed.io iteration 1
sitespeedio_1  | [2020-04-26 00:17:06] INFO: Got https://sitespeed.io analysed from Google Page Speed Insights
sitespeedio_1  | [2020-04-26 00:17:19] INFO: https://sitespeed.io 13 requests, backEndTime: 307ms, firstPaint: 772ms, firstVisualChange: 833ms, FCP: 772ms, DOMContentLoaded: 731ms, LCP: 772ms, Load: 736ms, speedIndex: 833, perceptualSpeedIndex: 833, contentfulSpeedIndex: 833, visualComplete85: 833ms, lastVisualChange: 833ms
sitespeedio_1  | [2020-04-26 00:17:19] INFO: Start collecting Lighthouse result for https://sitespeed.io iteration 1
sitespeedio_1  | [2020-04-26 00:17:49] INFO: Got https://sitespeed.io analysed with id 200426_2B_afe19a8a7ee2988691354d8571ba2a78 from http://localhost:4000
sitespeedio_1  | [2020-04-26 00:17:49] INFO: WebPageTest result at: http://localhost:4000/results.php?test=200426_2B_afe19a8a7ee2988691354d8571ba2a78
sitespeedio_1  | [2020-04-26 00:17:52] INFO: HTML stored in /sitespeed.io/sitespeed-result/sitespeed.io/2020-04-26-00-16-57
wpt_sitespeedio_1 exited with code 0
```

I can also post the webpagetest/agent logs although there's a lot more of them. Might be better off just running the compose file yourself.

**Comment by soulgalore** [comment]
Hi @dekryptic the errors are not from WebPageTest, its from Browsertime/sitespeed.io. `chrome not reachable` could be a mismatch in Chrome/Chromedriver version but since you run the container it should match. I'm thinking it could be running out of memory error on the machine?

**Comment by Myvryn** [comment]
I believe I've seen that error as well when I've run out of disc space on the machine.

Get Outlook for Android<https://aka.ms/ghei36>
________________________________

**Comment by dekryptic** [comment]
But after the first failed run it consistently runs without error. And if I remove the webpagehost config from the command I don't get any errors.

I'll take a look and see if there are any memory or disk space issues.

**Comment by tobbes4711** [comment]
hi, i have the same error.

iam running on t3.large sitespeed.io 12.5.1 (because 12.7.1 had an issue with my aliases)

after the first url is crawled i can see these lines over and over again:
"(Session info: chrome=81.0.4044.92)
    at Object.throwDecodedError (/usr/src/app/node_modules/selenium-webdriver/lib/error.js:550:15)
    at parseHttpResponse (/usr/src/app/node_modules/selenium-webdriver/lib/http.js:565:13)
    at Executor.execute (/usr/src/app/node_modules/selenium-webdriver/lib/http.js:491:26)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Driver.execute (/usr/src/app/node_modules/selenium-webdriver/lib/webdriver.js:700:17)
"
the run script has 3GB memory. the urls list has 67 entries by 3 iterations.

Disk is nearly "empty"
$df -h
/dev/nvme0n1p1   30G  5.3G   25G  18%

**Comment by soulgalore** [comment]
@tobbes4711 could you please file a bug for the alias? 12.7.0 had a bug with alias but that was fixed in 12.7.1 but I guess it isn't fully fixed? I file an issue for that I would really appreciate it so I can have a look!

For the throw decode error, if you increase the memory (https://www.sitespeed.io/documentation/sitespeed.io/docker/#increase-memory) do that help?

**Comment by soulgalore** [comment]
@dekryptic did you see anything? Any specific reason you run both WPT and sitespeed.io on the same host?

**Comment by tobbes4711** [comment]
For the throw decode error, if you increase the memory (https://www.sitespeed.io/documentation/sitespeed.io/docker/#increase-memory) do that help?
not really. the load average is still 4-6 and the loop.sh script stops working. i have changed now my t3.large to c5.large as you suggested.


the alias issue is fixed in 12.8.1. just tested some minutes ago.

**Comment by soulgalore** [comment]
I would separate WebPageTest and run that on another server. Closing, but let me know if there's still issues.
